server/utils/cloudFunctions/functions/countAdvertisersCloudFunctions/countAdvertisersHelpers.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Description: Helper functions for count advertisers cloud functions.
+ * Author: Robert Marcus
+ * Date: July 14, 2020
+ */
+
+const { assert } = require('../test/testConfig');
+const { firestoreFieldValue } = require('./countAdvertisersConfig');
+
+/**
+ * Assert `advertiser` is a string
+ * @param {string} advertiser as derived from snapshot
+ */
+function checkAdvertiser(advertiser) {
+  assert.typeOf(advertiser, 'string');
+}
+
+/**
+ * Assert `startDate` is a string and a valid iso date string.
+ * If `startDate` is not a valid iso string, `isoStartDate` will == NaN.
+ * @param {string} startDate as derived from snapshot
+ * @param {Object} isoStartDate Date(startDate) output, json string
+ */
+function checkStartDate(startDate, isoStartDate) {
+  assert.typeOf(startDate, 'string');
+  assert.ok(isoStartDate.getFullYear()); 
+}
+
+/**
+ * Returns a filepath reference to a doc containing the numberOfAds an 
+ * advertiser released in a particular year. Will link to whatever year 
+ * the ad in the `snapshot` is from. 
+ * If `advertiser` or `startDate` field is not well-typed/formed, 
+ * an assertion error will be thrown.     
+ * @param {Object} snapshot a json string
+ * @param {Object} collection reference to a particular firestore collection
+ * @returns {Object} 
+ */
+function getAdvertiserCountReference(snapshot, collection) {
+  const data = snapshot.data();
+  const advertiser = data.advertiser;
+  const startDate = data.startDate;
+  const isoStartDate = new Date(startDate);
+
+  checkAdvertiser(advertiser);
```

Comment:
What is the purpose of these two methods?  If all you're doing is reading, you shouldn't have to validate your output -- validation is done at write time to ensure that the DB is always in a valid state.

Reply:
Will leave validation to write time only.